load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java

```diff
@@ -15,124 +15,238 @@
+      description = "The network address of the Kafka broker."
+  )
+  private String broker;
+  @Parameter(
+      names = {"--request_rate"},
+      description = "The rate at which each client will make requests."
+  )
+  private int requestRate = 10;
+  @Parameter(
+      names = {"--max_outstanding_requests"},
+      description = "The maximum number of outstanding requests each client will allow."
+  )
+  private int maxOutstandingRequests = 20;
+  @Parameter(
+      names = {"--burn_in_duration_seconds"},
+      description = "The duration, in seconds, to run without recording statistics to allow tuning."
+  )
+  private int burnInDurationSeconds = 20;
+  @Parameter(
+      names = {"--number_of_messages"},
+      description = "The total number of messages to publish in the test. Enabling this will " +
+          "override --loadtest_length_seconds. Enabling this flag will also enable the check for " +
+          "message loss. If set less than 1, this flag is ignored."
+  )
+  private int numberOfMessages = 0;
 
   public static void main(String[] args) {
     // Turns off all java.util.logging.
     LogManager.getLogManager().reset();
+    Driver driver = new Driver();
+    JCommander jCommander = new JCommander(driver, args);
+    if (driver.help) {
+      jCommander.usage();
+      return;
+    }
+    driver.run();
+  }
+
+  private void run() {
     try {
-      // Parse command line arguments.
-      BaseArguments baseArgs = new BaseArguments();
-      CPSArguments cpsArgs = new CPSArguments();
-      KafkaArguments kafkaArgs = new KafkaArguments();
-      CompareArguments dataComparisonArgs = new CompareArguments();
-      JCommander jCommander = new JCommander(baseArgs);
-      jCommander.addCommand(CPSArguments.COMMAND, cpsArgs);
-      jCommander.addCommand(KafkaArguments.COMMAND, kafkaArgs);
-      jCommander.addCommand(CompareArguments.COMMAND, dataComparisonArgs);
-      jCommander.parse(args);
-      if (jCommander.getParsedCommand() == null) {
-        if (baseArgs.isHelp()) {
-          jCommander.usage();
-        }
-        return;
+      Preconditions.checkArgument(
+          cpsPublisherCount > 0 ||
+              cpsSubscriberCount > 0 ||
+              kafkaPublisherCount > 0 ||
+              kafkaSubscriberCount > 0
+      );
+      Preconditions.checkArgument(
+          broker != null || (kafkaPublisherCount == 0 && kafkaSubscriberCount == 0));
+      Map<ClientParams, Integer> clientParamsMap = new HashMap<>();
+      clientParamsMap.putAll(ImmutableMap.of(
+          new ClientParams(ClientType.CPS_GCLOUD_PUBLISHER, null), cpsPublisherCount,
+          new ClientParams(ClientType.KAFKA_PUBLISHER, null), kafkaPublisherCount,
+          new ClientParams(ClientType.KAFKA_SUBSCRIBER, null), kafkaSubscriberCount
+      ));
+      // Each type of client will have its own topic, so each subscription will get
+      // cpsSubscriberCount subscribers cumulatively among each of the subscriptions.
```

Comment:
I don't really understand this sentence. It's trying to say that there are cpsSubscriptionFanout subscriptions, and each subscription will have 'cpsSubscriberCount / cpsSubscriptionFanout' subscribers pulling from it, right?